src/SLCore/Analysis/SLCoreAnalyzer.cs

```diff
@@ -57,32 +61,45 @@ public bool IsAnalysisSupported(IEnumerable<AnalysisLanguage> languages)
 
     public void ExecuteAnalysis(string path, Guid analysisId, IEnumerable<AnalysisLanguage> detectedLanguages, IIssueConsumer consumer,
         IAnalyzerOptions analyzerOptions, CancellationToken cancellationToken)
-    {   
+    {
         var analysisStatusNotifier = analysisStatusNotifierFactory.Create(nameof(SLCoreAnalyzer), path);
         analysisStatusNotifier.AnalysisStarted();
-        
+
         var configurationScope = activeConfigScopeTracker.Current;
         if (configurationScope is not { IsReadyForAnalysis: true })
         {
             analysisStatusNotifier.AnalysisNotReady(SLCoreStrings.ConfigScopeNotInitialized);
             return;
         }
-        
+
         if (!serviceProvider.TryGetTransientService(out IAnalysisSLCoreService analysisService))
         {
             analysisStatusNotifier.AnalysisFailed(SLCoreStrings.ServiceProviderNotInitialized);
             return;
         }
-        
-        ExecuteAnalysisInternalAsync(path, configurationScope.Id, analysisId, analyzerOptions, analysisService, analysisStatusNotifier, cancellationToken).Forget();
+
+        Dictionary<string, string> extraProperties = [];
+        if (detectedLanguages != null && detectedLanguages.Contains(AnalysisLanguage.CFamily))
+        {
+            var compilationDatabasePath = compilationDatabaseLocator.Locate();
+            if (compilationDatabasePath == null)
+            {
+                analysisStatusNotifier.AnalysisFailed(SLCoreStrings.CompilationDatabaseNotFound);
```

Comment:
Hmm, this also means that the file will not be analyzed for secrets. Is that a desired behavior in this case?